utils/fix_cmocka_xml.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+'''
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+
+'''
+import subprocess
+import os
+
+# Some fixes for the CMOCKA headers
+xml_header ="<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\n"
+parent_header = "<data>\n"
+parent_footer = "</data>"
+path = os.path.join(os.path.dirname(os.path.realpath(__file__)),"../test_results/*.xml")
+cmd = "ls {}".format(path)
+out = subprocess.check_output(cmd,shell=True)
+# This is done because some XML files are not formed correctly
+# by CMOCKA framework. Some files doesn't have <xml..> or
+# <root> having nested tags [eg: if one group test called
+# repeatedly].
+# If there is a fix by CMOCKA framework, this is not required.
```

Comment:
You should document the upstream ticket you opened at cmocka and/or the ticket I opened if it's relevant.

Reply:
https://github.com/clibs/cmocka/issues/14 : CMOCKA ticket.

Reply:
You should put the ticket link right in the comment here in this file for future readers.

Reply:
https://github.com/clibs/cmocka/ is not the upstream cmocka git project.  https://cmocka.org/#git refers to https://gitlab.com/groups/cmocka.

You should reopen your issue at the above project and change the link here.